test/modules/aws-elasticache-integration.ts

```diff
@@ -38,26 +38,31 @@ const elasticacheclient = new ElastiCache({
   region,
 });
 
-const getAvailableNodeType = async () => {
+const getAvailableNodeTypes = async () => {
   const reservations = await elasticacheclient.describeReservedCacheNodesOfferings({});
   if (reservations && reservations.ReservedCacheNodesOfferings) {
     const items: string[] = [];
-    // iterate over list and get the ones matching the product description
+    // iterate over list and get the ones matching the product description, and small size
     reservations.ReservedCacheNodesOfferings.forEach(function (node) {
-      if (node.ProductDescription == cacheType) {
+      if (
+        node.ProductDescription == cacheType &&
+        (node.CacheNodeType?.includes('small') || node.CacheNodeType?.includes('micro'))
+      ) {
         if (node.CacheNodeType) items.push(node.CacheNodeType);
       }
     });
-    return items.pop() ?? '';
+    return items ?? [];
   }
-  return '';
+  return [];
 };
-let nodeType: string;
+let nodeType: string, updatedNodeType: string;
 
 jest.setTimeout(620000);
 beforeAll(async () => {
   // just sleep
-  nodeType = await getAvailableNodeType();
+  const nodes = await getAvailableNodeTypes();
+  nodeType = nodes.pop() ?? '';
+  updatedNodeType = nodes.pop() ?? '';
   await execComposeUp();
 }, 60000);
```

Comment:
this timeout seems off, I do not see it in other tests

Reply:
this came from experimentation with async. I guess it could be removed. The main issue i had is that those values were not recognized, just were taken as `undefined` . The `done` trick made them work, and values properly picked